### PR TITLE
Add a confirmation for harvesting Replica Pods with associated ckeys/minds that aren't ready to be revived.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Tweak a code path to do an early return.

Add an alert box before the user harvests a Replica Pod if that Replica Pod is going to give you seeds but it has a mind or ckey associated with it.

image

[Edit] - Additionally, I've protected against where two people try to harvest at once and spamming the crap outta the hydro trays to bring up a dozen prompts. A successful harvest will result in the seeds being qdel'd and thus a simple QDELETED check after the confirmation prompt is accepted should suffice to guard against any weirdness.

image

## Why It's Good For The Game

Botanists gave me feedback that it sucks to harvest a replica pod of a person before they're dead, either through accidentally clicking, not realising the replica pod target is still alive or even though other well-meaning players harvesting them and thinking they're helping a revival.

Now there's no ambiguity. If these seeds have a mind or ckey associated with them, a simple confirmation box makes sure the player REALLY wants to harvest seeds instead of waiting for a fully formed pod person.

## Changelog
:cl:
tweak: Added a confirmation prompt before harvesting a fully grown Replica Pod that has been associated with a player, but the player is not yet dead or ready to be revived. Gives you the choice of backing out or harvesting them for seeds. Functionality unchanged.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/replicapod.dm [hunk 1]
```diff
@@ -101,28 +101,13 @@
...
-			features["mcolor"] = "#59CE00"
-		for(var/V in quirks)
-			new V(podman)
-		podman.hardset_dna(null,null,null,podman.real_name,blood_type, new /datum/species/pod,features)//Discard SE's and UI's, podman cloning is inaccurate, and always make them a podman
-		podman.set_cloned_appearance()
-		log_cloning("[key_name(mind)] cloned as a podman via [src] in [parent] at [AREACOORD(parent)].")
+	// No podman player, give one or two seeds.
+	if(!make_podman)
+		// Prevent accidental harvesting. Make sure the user REALLY wants to do this.
+		var/choice = alert("The pod is currently devoid of soul. There is a possibility that a soul could claim this creature, or you could harvest it for seeds.", "Harvest Seeds?", "Harvest Seeds", "Cancel")
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/replicapod.dm:107]: what happens if you queue this prompt up a bunch and/or stall it?

**Comment by Winterous** [comment]
This pleases me! Can you also make it alert the dead player, like, when you're getting defibbed, if they're currently not ghosted from their corpse? Not really critical but would be a nice touch.

**Comment by Fikou** [comment]
cant you metagame who is dead with this?

**Comment by Timberpoes** [comment]
> 
> 
> cant you metagame who is dead with this?

Thankfully not. This prompt appears with the same message whether the player is dead or alive, whether they've suicided or are indeed revivable.

I didn't want to add any additional metagame information if I could avoid it. The only thing this tells you is that the seeds used to make the pod person have been injected with some blood and that blood belonged to a mob that had a mind or ckey associated with it.

**Comment by Timberpoes** [comment]
> 
> 
> This pleases me! Can you also make it alert the dead player, like, when you're getting defibbed, if they're currently not ghosted from their corpse? Not really critical but would be a nice touch.

I can definitely look into this in the future.

File: code/modules/hydroponics/grown/replicapod.dm (hunk 1 above)
Comment (Timberpoes) [reply code/modules/hydroponics/grown/replicapod.dm:107]: Good point. I'll remedy this shortly.

**Comment by ATH1909** [comment]
> > cant you metagame who is dead with this?
> 
> Thankfully not. This prompt appears with the same message whether the player is dead or alive, whether they've suicided or are indeed revivable.

You sure? Looks to me like the prompt only appears if harvesting the plant wouldn't make a podperson.

**Comment by Timberpoes** [comment]
> 
> 
> > > cant you metagame who is dead with this?
> > 
> > 
> > Thankfully not. This prompt appears with the same message whether the player is dead or alive, whether they've suicided or are indeed revivable.
> 
> You sure? Looks to me like the prompt only appears if harvesting the plant wouldn't make a podperson.

The prompt could appear for any of the following reasons:
They're not dead.
They're dead but can't re-enter corpse.
They're dead but it was because they suicided verbed.
They're dead but also hellbound from signing an infernal contect.

The pod person is right there, in the hydroponics pod. The player would, in-game, have all the information they need as to if their pod person is catatonic or not prior to attempting the harvest.

Technically on live you can gather all this same information by just attempting to harvest the pod person. If they're not dead, or are dead but have some restriction, you get seeds. Of course if you have a 100u beaker of their blood, you can repeat this as many times as necessary. If you don't, the chance at revival is gone unless you can get more of their blood.

This PR gives you exactly the same information you could get on live right now - But doesn't harvest the pod person into a seed or two in the process.

I feel adding this QoL feature is worth the tradeoff that a person doesn't need to keep injecting seeds with blood and hoping *this time* is the time the player is ready to be revived.

However, it's still up to another maintainer to approve this or request changes.

**Comment by Winterous** [comment]
Maybe it should be a delay instead of a confirmation? One time my podding failed because I entered my corpse the exact moment they harvested, because I wanted to check if my pod was ready yet and it was right next to my corpse. Or it could be both a confirmation and a delay?
Only if it has blood injected of course, if it's a seed-pod then no delay.

**Comment by ExcessiveUseOfCobblestone** [comment]
looks like I could queue the alert and activate it across the map

**Comment by Timberpoes** [comment]
> 
> 
> looks like I could queue the alert and activate it across the map

Testing a quick fix for that now.

**Comment by Timberpoes** [comment]
Added a check to make sure the parent hydroponics bay is interactible after the alert box has an option selected.

**Comment by Timberpoes** [comment]
(I'm just using my own PR to test something, ignore the close/open)

**Pull request merged**